packages/mllp-server/src/parsing.ts

```diff
@@ -0,0 +1,56 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+
+import { Hl7Message } from "@medplum/core";
+import { getSendingApplication } from "@metriport/core/command/hl7v2-subscriptions/hl7v2-to-fhir-conversion/msh";
+import {
+  createUnpackPidFailureFileKey,
+  getCxIdAndPatientIdOrFail,
+  getOptionalValueFromMessage,
+} from "@metriport/core/command/hl7v2-subscriptions/hl7v2-to-fhir-conversion/shared";
+import { utcifyHl7Message } from "@metriport/core/external/hl7-notification/datetime";
+import { Logger } from "@metriport/core/util/log";
+import { buildDayjs, ISO_DATE_TIME } from "@metriport/shared/common/date";
+import { Config } from "./config";
+import { asString, bucketName, s3Utils } from "./utils";
+
+const hieTimezoneDictionary = Config.getHieTimezoneDictionary();
+
+export interface ParsedHl7Data {
+  message: Hl7Message;
+  cxId: string;
+  patientId: string;
+}
+
+export async function parseHl7Message(rawMessage: Hl7Message): Promise<ParsedHl7Data> {
+  const sendingApplication = getSendingApplication(rawMessage) ?? "Unknown HIE";
+  const hieTimezone = hieTimezoneDictionary[sendingApplication] ?? "UTC";
+  const message = utcifyHl7Message(rawMessage, hieTimezone);
+
+  const { cxId, patientId } = getCxIdAndPatientIdOrFail(message);
+
+  return {
+    message,
+    cxId,
+    patientId,
+  };
+}
+export async function handleParsingError(rawMessage: Hl7Message, logger: Logger) {
```

Comment:
This function might be used while handling errors, but it's goal is not really that, just storing the raw income message, without parsing, isn't it?

I'd expect a function that handles errors to get the error as a parameter. 😄 

BTW, that brings an idea: would it be useful to store the error on a sibling file in S3?